sophia/src/term/iri.rs

```diff
@@ -430,30 +481,65 @@ where
 //     }
 // }
 
-impl<TD> Hash for Iri<TD>
+impl<'a, T, U> From<&'a Iri<U>> for Iri<T>
 where
-    TD: TermData,
+    T: TermData + From<&'a str>,
+    U: TermData,
 {
-    fn hash<H: Hasher>(&self, state: &mut H) {
-        state.write(self.ns.as_ref().as_bytes());
-        state.write(self.suffix_as_str().as_bytes());
-        state.write_u8(0xff);
+    fn from(other: &'a Iri<U>) -> Self {
+        other.copy_with(T::from)
     }
 }
 
-impl<TD> fmt::Display for Iri<TD>
+impl<TD> Hash for Iri<TD>
 where
     TD: TermData,
 {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let ns = self.ns.as_ref();
-        let suffix = self.suffix_as_str();
-        write!(f, "{}{}", ns, suffix)
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        state.write(self.ns.as_ref().as_bytes());
+        state.write(self.suffix_as_str().as_bytes());
+        state.write_u8(0xff);
     }
 }
 
 #[cfg(test)]
 mod test {
+    use super::*;
+
+    #[cfg(debug_assertions)]
+    mod debug {
+        use super::*;
+
+        #[test]
```

Comment:
```suggestion
        #[cfg(debug)]
        #[test]
```
The behaviour tested by this function is only valid in debug mode.
If you ran `cargo test --release`, it would fail.